app/src/main/java/org/mozilla/fenix/ext/Activity.kt

```diff
@@ -48,3 +56,61 @@ fun Activity.breadcrumb(
         )
     )
 }
+
+/**
+ * Opens the set default browser setting depending on the user OS version:
+ * for devices with Q or lower takes the user to the settings page
+ * and for other allow the user to set the default browser in app.
+ */
+fun Activity.openSetDefaultBrowserOption() {
+    when {
+        Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q -> {
+            getSystemService(RoleManager::class.java).also {
+                if (it.isRoleAvailable(RoleManager.ROLE_BROWSER) && !it.isRoleHeld(
+                        RoleManager.ROLE_BROWSER
+                    )
+                ) {
+                    startActivityForResult(
+                        it.createRequestRoleIntent(RoleManager.ROLE_BROWSER),
+                        REQUEST_CODE_BROWSER_ROLE
+                    )
+                } else {
+                    navigateUserToDefaultAppsSettings()
+                }
+            }
+        }
+        Build.VERSION.SDK_INT >= Build.VERSION_CODES.N -> {
+            navigateUserToDefaultAppsSettings()
+        }
+        else -> {
+            (this as HomeActivity).openToBrowserAndLoad(
+                searchTermOrURL = SupportUtils.getSumoURLForTopic(
+                    this,
+                    SupportUtils.SumoTopic.SET_AS_DEFAULT_BROWSER
+                ),
+                newTab = true,
+                from = BrowserDirection.FromSettings
+            )
+        }
+    }
+}
+
+private fun Activity.navigateUserToDefaultAppsSettings() {
```

Comment:
Perhaps also add a kDoc for this. Can never have enough documentation to help people not have to read code.